### PR TITLE
Single code base for Python 2 and 3

### PR DESCRIPTION
**Pull request by andreabedini**
Work in progress. This has to be thoroughly checked. From the top of my head surely the `range/xrange` dichotomy will cause problems.

### CODE REVIEW

**Comment by scopatz** [comment]
This looks good to me - why do you think that the range/xrange things will cause problems?  Also do you want to do more work on this before merging?

**Comment by andreabedini** [comment]
@scopatz because Py3 range and Py2 xrange are not really equivalent (since int/long are different).

```
❯❯❯ python3 -c 'print(len(range(10**100, 10**100 + 10)))'
10
❯❯❯ python -c 'print(len(xrange(10**100, 10**100 + 10)))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
OverflowError: Python int too large to convert to C long
```

**Comment by andreabedini** [comment]
If we are ready to break things we might just merge it, bugfix can still come from the release branch.

**Comment by scopatz** [comment]
I am happy to merge this if you think that it is ready.

**Comment by scopatz** [comment]
Just say the word :)

File: tables/array.py [hunk 1]
```diff
@@ -566,7 +566,7 @@ def translate_slice(exp, length):
                     else:
                         list_seen = True
                 else:
-                    if (not isinstance(exp[0], (int, long, numpy.integer)) or
+                    if (not isinstance(exp[0], (int, int, numpy.integer)) or
```
Comment (avalentino) [inline tables/array.py]: Duplicate int:
```
   (int, long, numpy.integer) -->  (int, numpy.integer)
```

File: tables/expression.py [hunk 2]
```diff
@@ -177,8 +181,8 @@ def __init__(self, expr, uservars=None, **kwargs):
         self.names, _ = getExprNames(expr, context)
 
         # Raise a ValueError in case we have unsupported objects
-        for name, var in vars_.iteritems():
-            if type(var) in (int, long, float, str):
+        for name, var in six.iteritems(vars_):
+            if type(var) in (int, int, float, str):
```
Comment (avalentino) [inline tables/expression.py]: Duplicate int

File: tables/idxutils.py [hunk 3]
```diff
@@ -476,7 +477,7 @@ def nextafter(x, direction, dtype, itemsize):
     direction."""
 
     assert direction in [-1, 0, +1]
-    assert dtype.kind == "S" or type(x) in (bool, int, long, float)
+    assert dtype.kind == "S" or type(x) in (bool, int, int, float)
```
Comment (avalentino) [inline tables/idxutils.py]: Duplicate int

File: tables/table.py [hunk 4]
```diff
@@ -822,7 +826,7 @@ def __init__(self, parentnode, name,
 
         # Check the chunkshape parameter
         if new and chunkshape is not None:
-            if isinstance(chunkshape, (int, numpy.integer, long)):
+            if isinstance(chunkshape, (int, numpy.integer, int)):
```
Comment (avalentino) [inline tables/table.py]: Duplicate int

File: tables/vlarray.py [hunk 5]
```diff
@@ -325,7 +329,7 @@ def __init__(self, parentnode, name, atom=None, title="",
 
         # Check the chunkshape parameter
         if new and chunkshape is not None:
-            if isinstance(chunkshape, (int, numpy.integer, long)):
+            if isinstance(chunkshape, (int, numpy.integer, int)):
```
Comment (avalentino) [inline tables/vlarray.py]: Duplicate int

**Comment by avalentino** [comment]
Great idea @andreabedini !
I had no time to test it but I'm abolutely +1 for this change

**Comment by FrancescAlted** [comment]
This looks great to me too!

**Comment by andreabedini** [comment]
Now that I understand better the issues with `x/l/range` I am confident this should not cause too much breakage. So @scopatz, merge :shipit: !

**Comment by scopatz** [comment]
Thanks @andreabedini! Sorry about the delay. I am in the middle of proposal writing season :(

**Comment by andreabedini** [comment]
no need to apologise @scopatz!

**Comment by andreabedini** [comment]
btw I just passed you on the contributors list :P

**Comment by scopatz** [comment]
hahah Well deserved :)

**Pull request merged**